app/src/main/java/org/xbmc/kore/ui/sections/localfile/HttpApp.java

```diff
@@ -0,0 +1,113 @@
+package org.xbmc.kore.ui.sections.localfile;
+
+
+import android.app.Activity;
+import android.net.wifi.WifiManager;
+import android.util.Log;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Map;
+
+import fi.iki.elonen.NanoHTTPD;
+
+import static android.content.Context.WIFI_SERVICE;
+
+
+public class HttpApp extends NanoHTTPD {
+
+    public HttpApp(Activity activity, int port) throws IOException {
+        super(port);
+        this.port = port;
+        this.activity = activity;
+        start(NanoHTTPD.SOCKET_READ_TIMEOUT, false);
+    }
+
+    static int port;
+    static Activity activity;
```

Comment:
Neither of this should be static. Both should me regular members.
Do we need port here? Isn't it accessible via the superclass?
It probably should be an Activity but a Context.